### PR TITLE
[Pytorch][build] udpate OFI and EFA version

### PR DESCRIPTION
**Pull request by Qingzi-Lan**
### Description
update OFI and EFA package version to support new sagemaker training

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu [hunk 1]
```diff
@@ -102,10 +102,14 @@ RUN mkdir /tmp/efa \
   && curl -O https://s3-us-west-2.amazonaws.com/aws-efa-installer/aws-efa-installer-${EFA_VERSION}.tar.gz \
   && tar -xf aws-efa-installer-${EFA_VERSION}.tar.gz \
   && cd aws-efa-installer \
+  && apt-get update \
+  && apt-get upgrade -y \
```
Comment (ydaiming) [inline pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu:106]: Remove this line to avoid upgrading any pinged packages accidentally.

File: pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu [hunk 2]
```diff
@@ -62,7 +62,7 @@ RUN apt-get update \
     jq \
     libcublas-11-3=${CUBLAS_VERSION}-1 \
     libcublas-dev-11-3=${CUBLAS_VERSION}-1 \
-    libcudnn8=$CUDNN_VERSION-1+cuda11.3 \
```
Comment (ydaiming) [inline pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu:65]: revert and still ping the `libcudnn8` version.

File: pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu [hunk 3]
```diff
@@ -102,10 +102,14 @@ RUN mkdir /tmp/efa \
...
   && tar -xf aws-efa-installer-${EFA_VERSION}.tar.gz \
   && cd aws-efa-installer \
+  && apt-get update \
+  && apt-get upgrade -y \
   && ./efa_installer.sh -y --skip-kmod -g \
   && rm -rf $OPEN_MPI_PATH \
   && rm -rf /tmp/efa \
-  && rm -rf /tmp/aws-efa-installer-${EFA_VERSION}.tar.gz
+  && rm -rf /tmp/aws-efa-installer-${EFA_VERSION}.tar.gz \
+  && rm -rf /var/lib/apt/lists/* \
```
Comment (ydaiming) [inline pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu:111]: Let's also add these two lines to all `apt-get update` commands.
```
  && rm -rf /var/lib/apt/lists/* \
  && apt-get clean
```

File: pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu [hunk 4]
```diff
@@ -226,7 +231,9 @@ RUN apt-get update \
  && cat /etc/ssh/ssh_config | grep -v StrictHostKeyChecking > /etc/ssh/ssh_config.new \
  && echo "    StrictHostKeyChecking no" >> /etc/ssh/ssh_config.new \
  && mv /etc/ssh/ssh_config.new /etc/ssh/ssh_config \
- && rm -rf /var/lib/apt/lists/*
+ && rm -rf /var/lib/apt/lists/* \
```
Comment (ydaiming) [inline pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu:234]: nit: repeated command

**Review by ydaiming — APPROVED** [review]

File: pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu [hunk 5]
```diff
@@ -102,10 +102,13 @@ RUN mkdir /tmp/efa \
   && curl -O https://s3-us-west-2.amazonaws.com/aws-efa-installer/aws-efa-installer-${EFA_VERSION}.tar.gz \
   && tar -xf aws-efa-installer-${EFA_VERSION}.tar.gz \
   && cd aws-efa-installer \
+  && apt-get update \
```
Comment (junpuf) [inline pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu:105]: what this update is used for? are there repo information needs to be refreshed?

Comment (Qingzi-Lan) [reply pytorch/training/docker/1.10/py3/cu113/Dockerfile.e3.gpu:105]: we have `apt-get clean` in line 92 for reducing image size. this line is to get repo info again

**Review by junpuf — APPROVED** [review]
LGTM
